File: Solution_125_valid_palindrome.java
public class Solution{
    public boolean isPalindrome(String s) { //12345  1234
        if(s.isEmpty()||s.length()<1) return true;
        
        s = s.trim().toLowerCase();
        s = s.replaceAll("[^a-z0-9]", "");
        System.out.println(s);

        // Math.ceil(s.length()/2.0)
        for(int i=0; i< s.length()/2; i++){
            if(s.charAt(i)!= s.charAt(s.length()-1-i)){
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        Solution slt = new Solution();
        System.out.println(
            slt.isPalindrome(new String("@##@ 123 2 1"))
        );
    }
}

File: Solution_125_valid_palindrome.py
import re

class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """

        s = s.strip().lower()
        s = re.sub(r'\W+', '', s)

        print s

        for i in range(0, len(s)/2):
            if s[i]!=s[len(s)-1-i]:
                return False;

        return True;


slt = Solution();
res = slt.isPalindrome("@##@ 122 2 1 \t 2221")

print res
